### PR TITLE
osd: add log information to record the cause of do_osd_ops failure

### PR DESCRIPTION
**Pull request by NancySu05**
function
add log information to prepare_transaction function when do_osd_ops
failed, because do_osd_ops function body is too long(about 1770+ lines) 
to locate a problem.

Fixes: http://tracker.ceph.com/issues/41210
Signed-off-by: NancySu05 su_nan@inspur.com

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by NancySu05** [comment]
> 
> 
> did you intend to delete src/test/common/test_blkdev_sys_block/sys/dev/block/*?

no，i just mean to commit PrimaryLogPG.cc

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -8328,6 +8328,7 @@ int PrimaryLogPG::prepare_transaction(OpContext *ctx)
   // prepare the actual mutation
   int result = do_osd_ops(ctx, *ctx->ops);
   if (result < 0) {
+    dout(10) << " do_osd_ops error: " << cpp_strerror(result) << dendl;
```
Comment (tchaikov) [inline src/osd/PrimaryLogPG.cc]: i'd suggest move this into `PrimaryLogPG::do_osd_ops()`.

Comment (NancySu05) [reply src/osd/PrimaryLogPG.cc]: i'v recommitted follow your advice, please review.

**Comment by NancySu05** [comment]
how to reopen this PR?

**Comment by sebastian-philipp** [comment]
you cannot. please create a new PR

**Comment by NancySu05** [comment]
> you cannot. please create a new PR

I have created a new PR  #29787,  please review, thanks.

**Pull request closed without merging**